mono/mini/method-to-ir.c

```diff
@@ -9139,6 +9140,8 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 
 			/* INSTANCE CASE */
 
+			if (is_instance)
+				g_assert (field->offset);
```

Comment:
`g_assert (!is_instance || field->offset);`

Reply:
So the difference between the two is that the if version is easy to understand, while the other version requires some thinking.

Reply:
This basic boolean logic, alas.